File: we_3_6_linearizing_a_range_measurement.m
% Integrated Navigation Systems — Chapter 3: Deterministic and Stochastic Foundations
% Worked Example 3.6 — Linearizing a range measurement
% Companion code (educational reference implementation; see the book for context).

p=[3;4]; pb=[0;0]; r=norm(p-pb);
H=((p-pb)/r)';  fprintf('H=[%.1f %.1f]\n',H);   % [0.6 0.8]

File: we_3_6_linearizing_a_range_measurement.py
# Integrated Navigation Systems — Chapter 3: Deterministic and Stochastic Foundations
# Worked Example 3.6 — Linearizing a range measurement
# Companion code (educational reference implementation; see the book for context).

import numpy as np
p=np.array([3.,4]); pb=np.zeros(2); r=np.linalg.norm(p-pb)
H=(p-pb)/r;  print('H=',np.round(H,1))   # [0.6 0.8]
